### PR TITLE
Dev content tree

### PR DESCRIPTION
**Pull request by skurfuerst**

### CODE REVIEW

File: Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import {Tree} from 'Host/Components/';
+import {immutableOperations} from 'Shared/Util';
+import backend from 'Host/Service/Backend.js';
+import {connect} from 'react-redux';
+
+const {$immutable, $get, $set} = immutableOperations;
+
+
+let buildTreeData;
+buildTreeData = (nodes, nodePaths) => {
```
Comment (dimaip) [inline Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js]: Did you think of using reselect selector for this? Would improve performance and keep concerns a bit more separated. You can think of reselect as of Q in CQRS.

File: Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+export default class ContentTree extends Component {
+
+    static propTypes = {
+        collapsedNodes: PropTypes.instanceOf(Immutable.List),
+        nodes: PropTypes.instanceOf(Immutable.Map),
+        thisDocumentNodePath: PropTypes.string
+    }
+
+    render() {
+        const treeData = buildTreeData(this.props.nodes, (this.props.thisDocumentNodePath ? [this.props.thisDocumentNodePath] : []));
```
Comment (dimaip) [inline Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js]: In any case it's better to move this out to @connect. React component itself should stay as pure as possible and not be concerned with this selector logic here.

File: Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                data={treeData}
+                onNodeToggle={this.onPageNodeToggle.bind(this)}
+                onNodeClick={this.onPageNodeClick.bind(this)}
+                onNodeFocusChanged={this.onPageNodeFocusChanged.bind(this)}
+                />
+        );
+    }
+
+    onPageNodeToggle(node) {
+        const {nodeTreeService} = backend;
```
Comment (dimaip) [inline Resources/Private/JavaScript/Host/Containers/LeftSideBar/ContentTree/index.js]: Just FYI: I've heard that service layer is being refactored. Also note this: https://github.com/PackageFactory/PackageFactory.Guevara/issues/23

**Comment by Inkdpixels** [comment]
Hey @skurfuerst - I've updated the structure of the initialState/reducers/actions and included unit tests. Could you rebase this PR to the latest master and include unit tests for your added actions?

Thanks :)

**Comment by skurfuerst** [comment]
closing, outdated.

**Pull request closed without merging**